List item version history

Atmire-Kristof opened the pull request:
This PR adds item version histories to the item pages (simple and full), and a separate page to manage the versions, in a similar manner to how this is done in DSpace 6.

### Item pages
If an item contains a version history with at least 1 version, a list will be displayed at the bottom of their simple and full item pages.
This list displays all the versions within the item's version history with 5 columns:
- Version: The version's number
- Item: The handle of the item this version represents (in the form of a link to their item page) with an asterisk behind the handle for the version of the item's page you're currently on.
- Editor: The name of the E-Person that created the version (in the form of a `mailto` link to the eperson's email)
- Date: The date this version was created
- Summary: The summary of the version

The list of versions is also paginated with a default page-size of 10. Pagination will be hidden if there's only one page available.
The pagination parameters also do not affect any existing URL parameters. This is to avoid the version list's pagination from interfering with other pagination on the item page (like relationship search).

### Edit Item Version History
Added a new option under "Edit" in the admin sidebar named "Item Version History".
This opens up a modal similar to other sidebar edit options, where the user can select an item to edit their version history for (automatically lists the item you're visiting the item page for if opened on an item page).
Selecting an item will open up a new page on route `/items/<item-uuid>/edit/versionhistory`. This page displays the item's version history similar to the item pages (re-uses the same component). The only difference being that it displays an alert when the item's version history is empty (instead of hiding the whole component) and a return button to return to the item page.
Currently, this page only lists versions within the version history. It isn't possible to modify the version history yet, as this is not yet supported in the REST API.

### Example
<img width="1124" alt="Screenshot 2020-02-21 at 15 04 39" src="https://user-images.githubusercontent.com/34308852/75040753-89753980-54bb-11ea-9cb1-e59243e06679.png">

tdonohue commented on src/app/+item-page/item-page.resolver.ts:
```diff
@@ -28,6 +28,7 @@ export class ItemPageResolver implements Resolve<RemoteData<Item>> {
       followLink('owningCollection'),
       followLink('bundles'),
       followLink('relationships'),
+      followLink('version', undefined, followLink('versionhistory')),
```
I'm getting an error on this line when I try to run this PR with AoT compilation (`yarn start`) or via `yarn run watch`.  Here's the error:
```
ERROR in app/+item-page/item-page.resolver.ts:31:40 - error TS2345: Argument of type 'FollowLinkConfig<HALResource>' is not assignable to parameter of type 'boolean'.

31       followLink('version', undefined, followLink('versionhistory')),
```

artlowel replied:
@tdonohue This is due to #588 that was just merged. I'll let @Atmire-Kristof know

Atmire-Kristof commented:
@tdonohue Thanks for the feedback. I've applied a couple of changes you suggested:
- Removed the sidebar option
- Moved the "Edit Version History" to the "Item Edit" page instead under its own tab (this also fixes the html title)
- Added a warning notice on top of the "Edit Version History" tab saying "Editing or adding new versions is not yet possible in this user interface"

tdonohue commented:
This was flagged for one approval and has been open for several weeks.  Merging immediately as it now works for me.
